FileNotFoundException: Could not load file or assembly 'System.ComponentModel.Composition on Linux

### Is there an existing issue for this?

- [x] I have checked the existing issues for an existing issue. I even actually looked at the ones that are pinned to the top!

### Operating System

Linux Mint

### CKAN Version

Latest nightly as of posting

### Games

KSP 1

### Game Version

N/A

### Did you make any manual changes to your game folder (i.e., not via CKAN)?

Deleted folders for individual mods I was no longer using, none of which are CKAN folders. Steam reported full success on verifying file integrity.

### Describe the bug

Running `ckan` in shell results in the following exception:
```
Unhandled Exception:
System.IO.FileNotFoundException: Could not load file or assembly 'System.ComponentModel.Composition, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies.
File name: 'System.ComponentModel.Composition, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
  at Autofac.Core.Registration.DefaultRegisteredServicesTracker.PopulateServiceInfo (Autofac.Core.Service service, Autofac.Core.Registration.ServiceRegistrationInfo info, System.Boolean isScopeIsolatedService) [0x00040] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Core.Registration.DefaultRegisteredServicesTracker.GetInitializedServiceInfo (Autofac.Core.Service service) [0x00058] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Core.Registration.DefaultRegisteredServicesTracker.ServiceRegistrationsFor (Autofac.Core.Service service) [0x00014] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Core.Registration.ComponentRegistry.ServiceRegistrationsFor (Autofac.Core.Service service) [0x00000] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Builder.StartableManager.StartStartableComponents (System.Collections.Generic.IDictionary`2[TKey,TValue] properties, Autofac.IComponentContext componentContext) [0x00028] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.ContainerBuilder.Build (Autofac.Builder.ContainerBuildOptions options) [0x00054] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at CKAN.ServiceLocator.Init () [0x0006c] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at CKAN.ServiceLocator.get_Container () [0x00009] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x000fa] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.IO.FileNotFoundException: Could not load file or assembly 'System.ComponentModel.Composition, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies.
File name: 'System.ComponentModel.Composition, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
  at Autofac.Core.Registration.DefaultRegisteredServicesTracker.PopulateServiceInfo (Autofac.Core.Service service, Autofac.Core.Registration.ServiceRegistrationInfo info, System.Boolean isScopeIsolatedService) [0x00040] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Core.Registration.DefaultRegisteredServicesTracker.GetInitializedServiceInfo (Autofac.Core.Service service) [0x00058] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Core.Registration.DefaultRegisteredServicesTracker.ServiceRegistrationsFor (Autofac.Core.Service service) [0x00014] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Core.Registration.ComponentRegistry.ServiceRegistrationsFor (Autofac.Core.Service service) [0x00000] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.Builder.StartableManager.StartStartableComponents (System.Collections.Generic.IDictionary`2[TKey,TValue] properties, Autofac.IComponentContext componentContext) [0x00028] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at Autofac.ContainerBuilder.Build (Autofac.Builder.ContainerBuildOptions options) [0x00054] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at CKAN.ServiceLocator.Init () [0x0006c] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at CKAN.ServiceLocator.get_Container () [0x00009] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x000fa] in <f122d6f9f30f4c38a552eec67cbf6c1c>:0 
```

### Steps to reproduce

1. Enter "ckan" into terminal, on Linux Mint 22 "Wilma" with the latest ckan nightly version installed.

### Relevant log output

_No response_

--- a/GUI/Controls/ManageMods.Designer.cs
+++ b/GUI/Controls/ManageMods.Designer.cs
@@ -327,18 +327,16 @@ namespace CKAN.GUI
             //
             // ModGrid
             //
-            this.ModGrid.Dock = System.Windows.Forms.DockStyle.Fill;
             this.ModGrid.AllowUserToAddRows = false;
             this.ModGrid.AllowUserToDeleteRows = false;
             this.ModGrid.AllowUserToResizeRows = false;
+            this.ModGrid.AutoSizeRowsMode = System.Windows.Forms.DataGridViewAutoSizeRowsMode.DisplayedCellsExceptHeaders;
             this.ModGrid.BackgroundColor = System.Drawing.SystemColors.Window;
-            this.ModGrid.EnableHeadersVisualStyles = false;
+            this.ModGrid.CellBorderStyle = System.Windows.Forms.DataGridViewCellBorderStyle.None;
             this.ModGrid.ColumnHeadersDefaultCellStyle.BackColor = System.Drawing.SystemColors.Control;
             this.ModGrid.ColumnHeadersDefaultCellStyle.SelectionBackColor = System.Drawing.SystemColors.Control;
             this.ModGrid.ColumnHeadersDefaultCellStyle.ForeColor = System.Drawing.SystemColors.ControlText;
             this.ModGrid.ColumnHeadersDefaultCellStyle.Padding = new System.Windows.Forms.Padding(2, 2, 2, 2);
-            this.ModGrid.DefaultCellStyle.ForeColor = System.Drawing.SystemColors.WindowText;
-            this.ModGrid.CellBorderStyle = System.Windows.Forms.DataGridViewCellBorderStyle.None;
             this.ModGrid.ColumnHeadersHeightSizeMode = System.Windows.Forms.DataGridViewColumnHeadersHeightSizeMode.AutoSize;
             this.ModGrid.Columns.AddRange(new System.Windows.Forms.DataGridViewColumn[] {
             this.Installed,
@@ -356,6 +354,10 @@ namespace CKAN.GUI
             this.InstallDate,
             this.DownloadCount,
             this.Description});
+            this.ModGrid.DefaultCellStyle.ForeColor = System.Drawing.SystemColors.WindowText;
+            this.ModGrid.DefaultCellStyle.Padding = new System.Windows.Forms.Padding(2, 4, 2, 4);
+            this.ModGrid.Dock = System.Windows.Forms.DockStyle.Fill;
+            this.ModGrid.EnableHeadersVisualStyles = false;
             this.ModGrid.Location = new System.Drawing.Point(0, 111);
             this.ModGrid.Margin = new System.Windows.Forms.Padding(4, 5, 4, 5);
             this.ModGrid.MultiSelect = true;

--- a/GUI/Model/GUIMod.cs
+++ b/GUI/Model/GUIMod.cs
@@ -23,7 +23,8 @@ namespace CKAN.GUI
         private CkanModule       Mod                 { get; set; }
         public  CkanModule?      LatestCompatibleMod { get; private set; }
         public  CkanModule?      LatestAvailableMod  { get; private set; }
-        public  InstalledModule? InstalledMod        { get; private set; }
+        public  InstalledModule? InstalledMod => registry.InstalledModule(Identifier);
+        private readonly IRegistryQuerier registry;
 
         /// <summary>
         /// The module of the checkbox that is checked in the MainAllModVersions list if any,
@@ -146,7 +147,7 @@ namespace CKAN.GUI
                    incompatible, hideEpochs, hideV)
         {
             IsInstalled      = true;
-            InstalledMod     = instMod;
+            this.registry    = registry;
             selectedMod      = registry.GetModuleByVersion(instMod.identifier, instMod.Module.version)
                                ?? instMod.Module;
             InstallDate      = instMod.InstallTime;
@@ -183,6 +184,7 @@ namespace CKAN.GUI
                       bool  hideEpochs,
                       bool  hideV)
         {
+            this.registry  = registry;
             Identifier     = mod.identifier;
             IsAutodetected = registry.IsAutodetected(Identifier);
             DownloadCount  = repoDataMgr.GetDownloadCount(registry.Repositories.Values, Identifier);

--- a/Tests/GUI/Model/GUIMod.cs
+++ b/Tests/GUI/Model/GUIMod.cs
@@ -177,7 +177,15 @@ namespace Tests.GUI
             using (var cacheDir = new TemporaryDirectory())
             using (var cache    = new NetModuleCache(cacheDir))
             {
-                var sut     = new GUIMod(instMod, repoData, Registry.Empty(repoData),
+                var registry = new Registry(repoData,
+                                            new Dictionary<string, InstalledModule>()
+                                            {
+                                                { instMod.identifier, instMod },
+                                            },
+                                            new Dictionary<string, string>(),
+                                            new Dictionary<string, string>(),
+                                            new SortedDictionary<string, Repository>());
+                var sut     = new GUIMod(instMod, repoData, registry,
                                          inst.KSP.StabilityToleranceConfig, inst.KSP,
                                          cache, false, true, true);
                 var callbackCount = 0;

--- a/debian/control.in
+++ b/debian/control.in
@@ -3,7 +3,7 @@ Version: @VERSION@
 Architecture: all
 Section: utils
 Priority: optional
-Depends: mono-runtime (>=5.0), ca-certificates-mono, libmono-microsoft-csharp4.0-cil, libmono-system-net-http-webrequest4.0-cil, libmono-system-servicemodel4.0a-cil
+Depends: mono-runtime (>=5.0), ca-certificates-mono, libmono-microsoft-csharp4.0-cil, libmono-system-net-http-webrequest4.0-cil, libmono-system-servicemodel4.0a-cil, libmono-system-componentmodel-composition4.0-cil
 Maintainer: The CKAN authors <debian@ksp-ckan.space>
 Description: KSP-CKAN official client.
  Official client for the Comprehensive Kerbal Archive Network (CKAN).

